### PR TITLE
Spliting snap.stop logic to e2 and e3 parts: snap.stop/snap.state.stop (corresponds to ProduceE2 and ProduceE3)

### LINKED ISSUE #10318: `--snap.stop` analog for e3 files: `--snap.state.stop`

### PR DESCRIPTION
**Pull request by JkLondon**
split Produce flag to ProduceE2 and ProduceE3 flag. Also replaced a logic in freezing files in snapshot. Added produce flag and it's setter to an aggregator.  Closes #10318

### CODE REVIEW

**Comment by JkLondon** [comment]
Linked issue: https://github.com/ledgerwatch/erigon/issues/10318

File: eth/stagedsync/stage_snapshots.go [hunk 1]
```diff
@@ -449,26 +449,27 @@ func SnapshotsPrune(s *PruneState, cfg SnapshotsCfg, ctx context.Context, tx kv.
...
+			//TODO: initialSync maybe save files progress here
 
 			var minBlockNumber uint64
 
 			if cfg.snapshotUploader != nil {
 				minBlockNumber = cfg.snapshotUploader.minBlockNumber()
 			}
 
-			if s.CurrentSyncCycle.IsInitialCycle {
+			if s.CurrentSyncCycle.IsFirstCycle {
```
Comment (AskAlexSharov) [inline eth/stagedsync/stage_snapshots.go:472]: this part is not related to current PR. let's move it to another one

File: eth/stagedsync/exec3.go [hunk 2]
```diff
@@ -156,6 +157,8 @@ func ExecV3(ctx context.Context,
 	chainConfig, genesis := cfg.chainConfig, cfg.genesis
 	blocksFreezeCfg := cfg.blockReader.FreezingCfg()
 
+	agg.SetProduceMod(blocksFreezeCfg.ProduceE3)
```
Comment (AskAlexSharov) [inline eth/stagedsync/exec3.go:160]: please move it to place where agg is creating. `backend.go`

File: eth/ethconfig/config.go [hunk 3]
```diff
@@ -136,7 +137,8 @@ func init() {
 type BlocksFreezing struct {
 	Enabled        bool
 	KeepBlocks     bool // produce new snapshots of blocks but don't remove blocks from DB
-	Produce        bool // produce new snapshots
+	ProduceE2      bool // produce new snapshots
```
Comment (AskAlexSharov) [inline eth/ethconfig/config.go:140]: `produce new snapshots` -> `produce new blocks files`

**Review by AskAlexSharov — CHANGES_REQUESTED** [review]
my advise: add

File: eth/stagedsync/exec3.go (hunk 2 above)
Comment (JkLondon) [reply eth/stagedsync/exec3.go]: moved setting and checked all the functions for the correct produce setting

File: eth/ethconfig/config.go (hunk 3 above)
Comment (JkLondon) [reply eth/ethconfig/config.go]: done

**Review by AskAlexSharov — APPROVED** [review]

**Review by AskAlexSharov — APPROVED** [review]

**Review by AskAlexSharov — APPROVED** [review]
